"Embedded struct" and "Interface mut:" and "if xxx is Interface{" complex, weird bugs

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.2.4 356ccf2** 
**OS: ArchLinux** 

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
fn main(){
	mut ll := LinearLayout{}
	mut lv := ListView{}
	ll.add(lv)	
	ll.layout()
}

//----------------------------------

interface Container{
mut:
	layout()
}

interface Layoutable{
	get_pos() (int, int)  // 1)  works if comment out
mut:
	set_pos(int, int)  // 3)  build error if change to immutable
}

//----------------------------------

pub struct LayouterBase{
mut:
	layoutables []Layoutable
}

pub fn (mut lb LayouterBase) add(layoutable Layoutable){
	lb.layoutables << layoutable
}

pub fn (lb LayouterBase) get_pos() (int, int){
	return 0, 0
}

pub fn (mut lb LayouterBase) set_pos(x int, y int){} // 3)  build error if change to immutable
// pub fn (lb LayouterBase) set_pos(x int, y int){} 

//-----------------------------------------

pub struct LinearLayout{
	LayouterBase
}

pub fn (mut ll LinearLayout) layout(){
	for mut l in ll.layoutables{
		dump(l.type_name())
		if l is Container{ // BUG this does not work
			dump(l.type_name())
		}
	}
}

//-----------------------------------------

pub struct ListView{
	LayouterBase
}

pub fn (mut lv ListView) layout(){}

// 2)  works if implements these methods
// pub fn (lv ListView) get_pos() (int, int){ return 0, 0 }
// pub fn (mut lv ListView) set_pos(x int, y int){}
```
There are two cases that works.

The weird part of this bug is that it is checking `Container` (`if l is Container{`), but somehow changing the method of `Layoutable` (`// get_pos() (int, int) ` or adding two method to ListView) changes the result.


**What did you expect to see?**
```
[con.v:47] l.type_name(): ListView
[con.v:49] l.type_name(): ListView
```
 
**What did you see instead?**
```
[con.v:47] l.type_name(): ListView
```
And if change `set_pos()` to immutable, then build error
```
==================
/tmp/v_1000/con.8335708713218948281.tmp.c:1962: error: field not found: LayouterBasex
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
````
````c
static inline void main__ListView_set_pos_Interface_main__Layoutable_method_wrapper(main__ListView* lb, int x, int y) {
	main__LayouterBase_set_pos(lb->LayouterBasex, y);
}
````

`main__LayouterBase_set_pos(lb->LayouterBasex, y);` This is wrong, isn't it?

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -1293,14 +1293,7 @@ pub fn (mut t Table) complete_interface_check() {
 		if tsym.kind != .struct_ {
 			continue
 		}
-		info := tsym.info as Struct
 		for _, mut idecl in t.interfaces {
-			if idecl.methods.len > tsym.methods.len {
-				continue
-			}
-			if idecl.fields.len > info.fields.len {
-				continue
-			}
 			if idecl.typ == 0 {
 				continue
 			}

--- a/vlib/v/tests/struct_embedding_with_interface_test.v
+++ b/vlib/v/tests/struct_embedding_with_interface_test.v
@@ -1,0 +1,60 @@
+fn test_struct_embedding_with_interface() {
+	mut ll := LinearLayout{}
+	mut lv := ListView{}
+	ll.add(lv)
+	ret := ll.layout()
+
+	println(ret)
+	assert ret.count('ListView') == 2
+}
+
+interface Container {
+mut:
+	layout() string
+}
+
+interface Layoutable {
+	get_pos() (int, int)
+mut:
+	set_pos(int, int)
+}
+
+pub struct LayouterBase {
+mut:
+	layoutables []Layoutable
+}
+
+pub fn (mut lb LayouterBase) add(layoutable Layoutable) {
+	lb.layoutables << layoutable
+}
+
+pub fn (lb LayouterBase) get_pos() (int, int) {
+	return 0, 0
+}
+
+pub fn (mut lb LayouterBase) set_pos(x int, y int) {}
+
+pub struct LinearLayout {
+	LayouterBase
+}
+
+pub fn (mut ll LinearLayout) layout() string {
+	mut output := ''
+	for mut l in ll.layoutables {
+		dump(l.type_name())
+		output += '$l.type_name()\n'
+		if l is Container {
+			dump(l.type_name())
+			output += '$l.type_name()\n'
+		}
+	}
+	return output
+}
+
+pub struct ListView {
+	LayouterBase
+}
+
+pub fn (mut lv ListView) layout() string {
+	return ''
+}

